transactions/add_signer.cdc

```diff
@@ -1,15 +1,16 @@
-import DAOTreasury from "../DAOTreasury.cdc"
+import DAOTreasury from "../contracts/DAOTreasury.cdc"
+import TreasuryActions from "../contracts/TreasuryActions.cdc"
 
-// 2.
 transaction(additionalSigner: Address) {
+
+  let Treasury: &DAOTreasury.Treasury{DAOTreasury.TreasuryPublic}
   
   prepare(signer: AuthAccount) {
-    let treasury = signer.borrow<&DAOTreasury.Treasury>(from: DAOTreasury.TreasuryStoragePath)
+    self.Treasury = signer.borrow<&DAOTreasury.Treasury>(from: DAOTreasury.TreasuryStoragePath)
                     ?? panic("Could not borrow the DAOTreasury")
-    let manager = treasury.borrowManager()
-    manager.addSigner(signer: additionalSigner)
   }
   execute {
-    return
+    let action =  TreasuryActions.AddSigner(additionalSigner)
```

Comment:
I'm confused here, on the source code TreasuryActions.AddSigner, looks like calling the `AddSinger` function will actually execute and emit the `AddSignerActionExecuted` event. Why proposing the action after the action is executed? tagging @jacksonConrad here as well, please let me know if I misunderstood the flow.

Reply:
I followed the flow for TransferTokenToTreasury.
ProposeAction creates the multisign requirement, so the action cannot be executed without all the necessary signatures
If the action isn't proposed, signed and executed, the treasury won't be updated with the new signer so the assertion will fail

Reply:
So this part is actually creating a struct, and it's not executing an action
`let action =  TreasuryActions.AddSigner(additionalSigner)`
Because AddSigner is defined like this:
`pub struct AddSigner: MyMultiSig.Action`
And after when the action is proposed, signed and executed, the execute() function is called upon this defined action

Reply:
Got it! Thanks for the clarification, now it totally makes sense